include/envoy/http/original_ip_detection.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "envoy/common/pure.h"
+#include "envoy/config/typed_config.h"
+#include "envoy/http/header_map.h"
+#include "envoy/network/address.h"
+
+namespace Envoy {
+namespace Http {
+
+struct OriginalIPDetectionParams {
+  Http::RequestHeaderMap& request_headers;
```

Comment:
Think it's worth caling out here the headers are pre-sanitiation?
Should be clear since you can't do sanitization until you know if upstream was trusted which you can't do without determining client IP, but can't hurt to call out

Reply:
Was a comment added?

Reply:
Yes updated pushed.